### PR TITLE
Fix double fee checking

### LINKED ISSUE #1220: Allow merge when account has minimum balance + one tx fee
Reported by @stanford-scs 

> So I had an account with 20.00001 XLM, and tried to merge with another account.  The merge failed with insufficient funds.  That's kind of annoying.  Really any account should be able to merge even if it has below the minimum, with the transaction fee just deducted from the amount credited to the destination account.

### PR DESCRIPTION
**Pull request by vogel**
Fixes #1220

### CODE REVIEW

**Comment by MonsieurNicolas** [comment]
let's use #1387 to get master back in a clean state wrt clang-format

File: src/main/Config.h [hunk 1]
```diff
@@ -31,7 +31,7 @@ class Config : public std::enable_shared_from_this<Config>
     std::string expandNodeID(std::string const& s) const;
 
   public:
-    static const int CURRENT_LEDGER_PROTOCOL_VERSION = 8;
+    static const int CURRENT_LEDGER_PROTOCOL_VERSION = 9;
```
Comment (MonsieurNicolas) [inline src/main/Config.h]: please make this change in its own commit as it's very significant

File: src/transactions/TransactionFrame.cpp [hunk 2]
```diff
@@ -296,8 +296,13 @@ TransactionFrame::commonValid(SignatureChecker& signatureChecker,
         return false;
     }
 
+    auto balanceAfter =
```
Comment (MonsieurNicolas) [inline src/transactions/TransactionFrame.cpp]: add a comment to explain that fees are captured in a different step when applying

**Comment by vogel** [comment]
All suggestions implemented

**Comment by MonsieurNicolas** [comment]
looks good but it has conflicts with master probably due to clang-format

**Comment by vogel** [comment]
updated

**Comment by MonsieurNicolas** [comment]
r+ 53308debfada0fbadf6f40757a7ed547e50b1d3c

**Pull request merged**